plugins/node/instrumentation-undici/src/types.ts

```diff
@@ -51,18 +52,29 @@ export interface RequestHookFunction<T = UndiciRequest> {
   (span: Span, request: T): void;
 }
 
+export interface ResponseHookFunction<
+  RequestType = UndiciResponse,
+  ResponseType = UndiciResponse
+> {
+  (span: Span, request: RequestType, response: ResponseType): void;
+}
+
 export interface StartSpanHookFunction<T = UndiciRequest> {
   (request: T): Attributes;
 }
 
 // This package will instrument HTTP requests made through `undici` or  `fetch` global API
 // so it seems logical to have similar options than the HTTP instrumentation
-export interface UndiciInstrumentationConfig<RequestType = UndiciRequest>
-  extends InstrumentationConfig {
+export interface UndiciInstrumentationConfig<
+  RequestType = UndiciRequest,
+  ResponseType = UndiciResponse
+> extends InstrumentationConfig {
   /** Not trace all outgoing requests that matched with custom function */
   ignoreRequestHook?: IgnoreRequestFunction<RequestType>;
   /** Function for adding custom attributes before request is handled */
   requestHook?: RequestHookFunction<RequestType>;
+  /** Function called once response headers have been received */
+  responseHook?: ResponseHookFunction<RequestType, ResponseType>;
```

Comment:
need to also be added to [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-undici#undicifetch-instrumentation-options)